### PR TITLE
Admin portal revamp - API Category Edit functionality

### PR DESCRIPTION
**Pull request by chathuranga95**
### Description

Please find the implementation for the  API Category: Edit functionality.

> Note: Previously added code snippets for development purposes have been cleaned.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminOperations/APICategories/EditAPICategory.jsx [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+let apiCategory = {};
+let mhistory;
+
+const updateAPICategory = (
+    apiCategoryId,
+    name,
+    setUpdatingAPICategory,
+    intl,
+) => {
+    console.log(apiCategoryId, name, apiCategory['description']);
```
Comment (Krishanx92) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminOperations/APICategories/EditAPICategory.jsx:114]: is this needed?

Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminOperations/APICategories/EditAPICategory.jsx]: That line is removed with commit: https://github.com/wso2/carbon-apimgt/pull/8429/commits/26d9169b808a61521bcf811b038010b581484e75

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminOperations/APICategories/EditAPICategory.jsx [hunk 2]
```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminOperations/APICategories/EditAPICategory.jsx:2]: Change license year

Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminOperations/APICategories/EditAPICategory.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8429/commits/53ccb83192807d36ad78f18dea06f817dd818cc7

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminOperations/APICategories/APICategories.jsx [hunk 3]
```diff
@@ -168,24 +151,13 @@ export default function APICategories() {
...
-                    <FormattedMessage
-                        id='contents.main.heading.api.categories.add_new.todo.remove.this'
-                        defaultMessage='Delete all - for dev'
-                    />
-                </Button>
             </div>
-            {mgLabels.length > 0 ? (
+            {apiCategories.length > 0 ? (
                 <Paper className={classes.gatewayPaper}>
                     <Table>
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminOperations/APICategories/APICategories.jsx:162]: Lets use MUIDataTables

Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminOperations/APICategories/APICategories.jsx]: Done with commit: https://github.com/wso2/carbon-apimgt/pull/8429/commits/6b6ca7115f8482710035e6759423b77d12ec7f78

**Pull request merged**